test/Tests.Infrastructure/RavenTestBase.cs

```diff
@@ -423,7 +423,6 @@ protected virtual DocumentStore GetDocumentStore(Options options = null, [Caller
 
                     store.BeforeDispose += (sender, args) =>
                     {
-                        var realException = Context.GetException();
```

Comment:
It logs the explicitly handled exception and leads to missleading stacktraces such:
```
(Raven.Client.Exceptions.Database.DatabaseNotRelevantException) -----
   at Raven.Server.Documents.DatabasesLandlord.CreateDatabaseConfiguration(StringSegment databaseName, Boolean ignoreDisabledDatabase, Boolean ignoreBeenDeleted, Boolean ignoreNotRelevant, DatabaseRecord databaseRecord) in C:\Jenkins\workspace\Testsv52_Winx86_Debug\s\src\Raven.Server\Documents\DatabasesLandlord.cs:line 878
   at Raven.Server.Documents.DatabasesLandlord.CreateDatabaseConfiguration(StringSegment databaseName, Boolean ignoreDisabledDatabase, Boolean ignoreBeenDeleted, Boolean ignoreNotRelevant) in C:\Jenkins\workspace\Testsv52_Winx86_Debug\s\src\Raven.Server\Documents\DatabasesLandlord.cs:line 858
   at Raven.Server.Documents.DatabasesLandlord.CreateDatabase(StringSegment databaseName, Nullable`1 wakeup, Boolean ignoreDisabledDatabase, Boolean ignoreBeenDeleted, Boolean ignoreNotRelevant, String caller) in C:\Jenkins\workspace\Testsv52_Winx86_Debug\s\src\Raven.Server\Documents\DatabasesLandlord.cs:line 638
   at Raven.Server.Documents.DatabasesLandlord.TryGetOrCreateResourceStore(StringSegment databaseName, Nullable`1 wakeup, Boolean ignoreDisabledDatabase, Boolean ignoreBeenDeleted, Boolean ignoreNotRelevant, String caller) in C:\Jenkins\workspace\Testsv52_Winx86_Debug\s\src\Raven.Server\Documents\DatabasesLandlord.cs:line 594
   at StressTests.Rachis.SubscriptionFailoverWithWaitingChainsStress.DropSubscriptions(String databaseName, Int32 SubscriptionsCount, ValueTuple`2 cluster, CancellationToken token) in C:\Jenkins\workspace\Testsv52_Winx86_Debug\s\test\StressTests\Rachis\SubscriptionFailoverWithWaitingChainsStress.cs:line 453

```

Reply:
It's handled in your case. Although that's not necessary the case in general. What do you think?

Reply:
I think its bad idea to keep code that can lead to missleading stacktraces, also the test from the PR it introduced in (https://github.com/ravendb/ravendb/commit/8b86d663749dbac4130ebb386295630585a6db7a#diff-b04a0b9c920a4494538c549bfd8ebe446a15963e312187c1e9796d043db22cd0R34), didn't even fail so I am not sure if it shows the "real exception"

Reply:
@karmeli87? You remember the reason of introducing "real exception" here?

Reply:
Yes.
It might happened that we throw some exception and due to that we fail to dispose the test. So we will mask the original exception (the one that actually caused the issue) in favor of the failed to dispose one.

Reply:
IIRC there is even a test for it :)

Reply:
the test passes without the  "real exception"  as well :)

Reply:
So sometime it's good but sometimes not, right? From the point of view of tests stabilization it's not an important factor as I understand. It's rather a matter of showing the correct error. 

In that case I'd rather leave it as is and just apply the two other fixes we have here but to 4.2 branch.